### ISSUE TITLE
Encoding avro schema of type enum

### ISSUE THREAD

**Issue by rumanhassan** [linkedin/goavro#63]
HI

I know this is not the good place to ask this question. But as I couldn't find much other resources for goavro, so thought of asking it here 

I am trying to encode a schema whose one of the fields type is enum. Following is the schema 

``` golang
var schema = `{
  "type": "record",
  "name": "Event",
  "namespace": "com.avro",
  "fields": [
    {
      "name": "timestamp",
      "type": "long"
    },
    {
      "name": "category",
      "type":  {
        "namespace": "com.avro",
            "name": "EventCategory",
            "type": "enum",
            "symbols": ["OptionA", "OptionB", "OptionC", "OptionD"]
         }
    }
  ]
}
`
```

I was able to encode timestamp field, but couldn't figured out how to encode category field as its type is enum. Followed below steps

step1 : 

``` golang
record, err := goavro.NewRecord(goavro.RecordSchema(schema))
if err != nil {
    log.Error("NewRecord failed ->", err.Error())
    return fmt.Errorf("NewRecord failed -> %v", err.Error())
}
```

step2:

``` golang
err := record.Set("timestamp", int64(123))
if err != nil {
    log.Error("timestamp Set Error ->", err.Error())
    return fmt.Errorf("timestamp Error -> %v", err)
}
```

step3: Need to set category, couldn't able to figure out how to set it

any help or suggestion is really appreciated.

Thanks

**Comment by karrick** [comment]
@rumanhassan, the behavior you ran into is definitely a usability issue, if not a bug. I have resolved this issue, along with #71 earlier today. Please let me know if you continue to have this problem, and I will dive back in.

**Comment by rumanhassan** [comment]
Yes its working for me now, it was the usability issue.

Thanks
Ruman.
